### ISSUE TITLE
Upload custom components to Github releases

### ISSUE THREAD

**Issue by schollz** [goreleaser/goreleaser#1251]
I have a hook that generates a vendored source base for my code. I'd like to upload this tarball of code along with the binaries generated from goreleaser.  I have a hook that generates this a tarball that I need into the `dist` directory, but I'd like a way that I could specify to goreleaser to upload this extra piece to Github too. 

If this already exists, please let me know! Right now I'm using goreleaser to generate the tarball (via hooks) and then manually uploading it.

**Comment by caarlos0** [comment]
> If this already exists, please let me know! Right now I'm using goreleaser to generate the tarball (via hooks) and then manually uploading it.

It doesn't. 

Thanks for the suggestion :)

**Comment by schollz** [comment]
Hi @caarlos0 . Thanks for your amazing work on this great tool. I really appreciate your code and also how well your engagement has helped to make goreleaser integrated into so many Go workflows (mine and others)!

I have figured out a way to do this (I run a script do upload via the Github API after goreleaser finishes) so feel free to close this.

**Comment by caarlos0** [comment]
thanks @schollz :) 

will leave it here, maybe more people need it... 

if not, it will autoclose in a couple of days :)

Cheers!

**Comment by leighmcculloch** [comment]
I was just trying to figure out how to do this. I'm needing to upload an additional file to the GitHub release at the end, and I can write a curl command to upload it, but I can't figure out how to get the GitHub Release ID from goreleaser to use in that upload. Any tips @schollz ?

**Comment by schollz** [comment]
@leighmcculloch Here's what I did and its not elegant. I added to my `goreleaser.yml` a hook to generate the file to upload. Then after runnning `goreleaser release` I manually run [a script to upload it](https://github.com/schollz/croc/blob/master/src/install/upload-src-tarball.sh). 

The "Github Relase ID" is just the version. In my script above the version comes from grepping my source code, but you could easily do `git describe`. And you can follow the rest of that script to identify the right file, repo, etc. Similar to `goreleaser` the `GITHUB_TOKEN` environmental variable is used, so that doesn't need to be set in that script.

This script would be pretty easy to integrate into goreleaser, and I could take a stab at it if there is interest. I would suggest making a new class called "extras" which specifies the filenames of extra files to upload. The files themselves could be generated as the user specifies in "hooks" (which is what I currently do).

**Comment by leighmcculloch** [comment]
Turns out I no longer need this. I needed this to write a homebrew file to the releases assets. I know most folks have a `homebrew-tap` repo but I didn't want to have to give GitHub Actions a broad write access to all the repositories I have access to and I was looking for ways to push the homebrew file to another place to support installing the formula via the non-shorthand brew commands.

But as it turns out I found a way to trick Homebrew into reading the formula from my current repo by renaming my repo `gas` to `homebrew-gas` and then back again, and thankfully brew follows the redirects that GitHub sets up. I detailed it [here](https://leighmcculloch.com/posts/homebrew-taps-without-the-homebrew-hyphen-prefix/).

This still seems like a useful feature but I don't need it right now.

**Comment by flw-cn** [comment]
I encountered a similar problem. (#1310) I want to add a termux distribution to my project. Then I did this with a release post-hook. But I can't upload it to the GitHub release now.

Thanks for @schollz 's idea, it helped me. And I agree that this is very inelegant.

I hope that the `release` section can also have a configuration item similar to `files` in `archives`, so that I can also publish the manual of my project as a GitHub file alone.
